StellarDotnetSdk/Soroban/SorobanServer.cs

```diff
@@ -24,32 +30,48 @@ public class SorobanServer : IDisposable
     private readonly bool _internalHttpClient;
     private readonly Uri _serverUri;
 
+    /// <summary>
+    ///     Constructs a new instance that will interact with the provided URL.
+    /// </summary>
+    /// <param name="uri">URL of the Soroban RPC server.</param>
+    /// <param name="httpClient">HttpClient instance to use for requests.</param>
     public SorobanServer(string uri, HttpClient httpClient)
     {
         _httpClient = httpClient;
         _serverUri = new Uri(uri);
         _internalHttpClient = false;
     }
 
-    public SorobanServer(string uri)
-        : this(uri, CreateHttpClient())
+    /// <summary>
+    ///     Constructs a new instance that will interact with the provided URL.
+    /// </summary>
+    /// <param name="uri">URL of the Soroban RPC server.</param>
+    /// <param name="bearerToken">Bearer token in case the server requires it.</param>
+    public SorobanServer(string uri, string? bearerToken = null)
```

Comment:
Do we need this? We also already pass a Bearer Token in the API on a same way just trough passing in a HttpClient:
<img width="598" alt="image" src="https://github.com/user-attachments/assets/13b1baa0-a67a-4d00-bc69-e26e00489bdf">

Which we actually need to update to this

```csharp
    private static Server GetServer(StellarConfig config, HttpClient httpClient)
    {
        if (string.IsNullOrEmpty(config.ServerToken))
        {
            return new Server(config.ServerUrl);
        }
        httpClient = Server.CreateHttpClient();
        httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue(
            "Bearer",
            config.ServerToken);
        return new Server(config.ServerUrl, httpClient);
    }
```

Same can be applied on the SorobanServer object.

Reply:
What I was thinking is that in most situations, our users will simply create a default HttpClient and pass the token. So, why don't we just handle that part for them?

Reply:
IMO, CreateHttpClient has no special logic, it's not even much "shorter" than what is inside it.
```
return CreateHttpClient(new HttpClientHandler());
```
So there is no reason to keep it, that's why I've marked it `Obsolete`.

The same applies for `CreateHttpClient(HttpMessageHandler handler).`, as I don't see any practical uses for our users.

I actually have a better idea.
```
public SorobanServer(string uri, string? bearerToken = null)
{
    _serverUri = new Uri(uri);
    _httpClient = new DefaultHttpClient(bearerToken);
    _internalHttpClient = true;
}
```
```
public DefaultHttpClient(string? bearerToken = null)
{
    var assembly = Assembly.GetAssembly(GetType())!.GetName();
    DefaultRequestHeaders.Add("X-Client-Name", "stellar-dotnet-sdk");
    DefaultRequestHeaders.Add("X-Client-Version", assembly.Version!.ToString());
    if (!string.IsNullOrEmpty(bearerToken))
    {
        DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", bearerToken);
    }
}
```
`DefaultHttpClient` is reusable between `SorobanServer` and `Server`. `HttpMessageHandler` related things can be removed altogether, as we're just using the default constructor anyway. What are your thoughts?

Reply:
That's fine to me too. Maybe add the client name and client version as optional parameters too. And maybe set the name in our projects to `BeansApp`.